packages/core/upload/admin/src/components/AssetCard/AssetCardBase.tsx

```diff
@@ -19,7 +19,7 @@ import { Pencil, Trash } from '@strapi/icons';
 import { useIntl } from 'react-intl';
 import { styled } from 'styled-components';
 
-import { getTrad } from '../../utils';
+import { getTrad } from '../../utils/utils';
```

Comment:
Becuase we have an eslint rule that prevents to use index files in the project, I used the same rules we are using in the review workflow, it is because of that I changed all the index files to a more comprehensive name

Reply:
>Becuase we have an eslint rule that prevents to use index files in the project, I used the same rules we are using in the review workflow

We already had the rule or you added it?

You actually are still using an index file you just changed the name to `utils`. So you are sort of hacking your way around the eslint rule. 

The content of `utils/utils.ts` is an index of all the files in the folder `utils`. I couldn't think of a better name than `index` 😄 . 

If we do not want index files. Then I think we should remove the index file (whatever its name is) and import the utils from their respective file. For example:

`import { somFileUtil } from '../utils/file'`

---

Not suggesting you need to do it it here but with this approach, in my opinion, the utils should be grouped together in the file with multiple named exports to have

`import {fileUtilA, fileUtilB} from '../utils/file'`

Currently most (or all) util functions have a dedicated file

Reply:
We already add this rule and when I did a meeting to organize the work I received the suggestion to use the config files we used in the review-workflow plugin.
I agree that the index is needed in some part and I can add a rule on top of the utils file to prevent the error at least for that file. Are you ok If I avoid the rule just in the utils file?

Reply:
>I agree that the index is needed in some part and I can add a rule on top of the utils file to prevent the error at least for that file

Unfortunately I don't think that is a solution...

Is this an eslint rule for the frontend? Because in review workflows server I see many index files. I'm not sure review workflows is a good example since on the frontend there is only a single component and here for upload the frontend is much more involved. 

<img width="394" alt="Screenshot 2024-11-04 at 11 17 55" src="https://github.com/user-attachments/assets/e8c1ba58-fa6e-48b4-b434-3388d33a415b">

Reply:
it is not used just in the review-workflow admin, we did it in other packages like content manager

Reply:
That's good news because I was curious why review-workflows was specifically suggested for the eslint rule since we should be using the same rules everywhere in my opinion. 

I'm not debating the eslint rule. I'm debating the manner in which you are importing files. I am familiar with the codebase and this is the first time I am seeing imports like this. It is a pattern you are introducing that I do not agree with and I don't think we should merge.

I believe the eslint rule to avoid index files is to encourage a flattened file structure which is the opposite of an import like name/name.